validator-cli/src/ArbToEth/transactionHandler.ts

```diff
@@ -0,0 +1,191 @@
+   *
+   * @returns False if transaction is pending || not final || not made, else True.
+   */
+  public async checkTransactionStatus(trnxHash: string | null, contract: ContractType): Promise<number> {
+    let provider: JsonRpcProvider;
+    if (contract === ContractType.INBOX) {
+      provider = this.veaInboxProvider;
+    } else if (contract === ContractType.OUTBOX) {
+      provider = this.veaOutboxProvider;
+    }
+
+    if (trnxHash == null) {
+      return 0;
+    }
+
+    const receipt = await provider.getTransactionReceipt(trnxHash);
+
+    if (!receipt) {
+      // TODO: Add transaction pending timeout- redo transaction.
+      this.emitter.emit(BotEvents.TXN_PENDING, trnxHash);
+      return 1;
+    }
+
+    const currentBlock = await provider.getBlock("latest");
+    const confirmations = currentBlock.number - receipt.blockNumber;
+
+    if (confirmations >= this.requiredConfirmations) {
+      this.emitter.emit(BotEvents.TXN_FINAL, trnxHash, confirmations);
+      return 3;
+    } else {
+      this.emitter.emit(BotEvents.TXN_NOT_FINAL, trnxHash, confirmations);
+      return 2;
+    }
+  }
+
+  /**
+   * Challenge claim for this.epoch in VeaOutbox(ETH).
+   *
+   */
+  public async challengeClaim() {
+    this.emitter.emit(BotEvents.CHALLENGING);
+    if (!this.claim) {
+      throw new ClaimNotSetError();
+    }
+    if ((await this.checkTransactionStatus(this.transactions.challengeTxn, ContractType.OUTBOX)) > 0) {
+      return;
+    }
+    const { deposit } = getBridgeConfig(this.chainId);
+    const gasEstimate: bigint = await this.veaOutbox[
+      "challenge(uint256,(bytes32,address,uint32,uint32,uint32,uint8,address))"
+    ].estimateGas(this.epoch, this.claim, { value: deposit });
+    const maxFeePerGasProfitable = deposit / (gasEstimate * BigInt(6));
+
+    // Set a reasonable maxPriorityFeePerGas but ensure it's lower than maxFeePerGas
+    let maxPriorityFeePerGasMEV = BigInt(6667000000000); // 6667 gwei
+
+    // Ensure maxPriorityFeePerGas <= maxFeePerGas
+    if (maxPriorityFeePerGasMEV > maxFeePerGasProfitable) {
+      maxPriorityFeePerGasMEV = maxFeePerGasProfitable;
+    }
+
+    const challengeTxn = await this.veaOutbox[
+      "challenge(uint256,(bytes32,address,uint32,uint32,uint32,uint8,address))"
+    ](this.epoch, this.claim, {
+      maxFeePerGas: maxFeePerGasProfitable,
+      maxPriorityFeePerGas: maxPriorityFeePerGasMEV,
+      value: deposit,
+      gasLimit: gasEstimate,
+    });
+    this.emitter.emit(BotEvents.TXN_MADE, challengeTxn.hash, this.epoch, "Challenge");
+    this.transactions.challengeTxn = challengeTxn.hash;
+  }
+
+  /**
+   * Withdraw the challenge deposit.
+   *
+   */
+  public async withdrawChallengeDeposit() {
+    this.emitter.emit(BotEvents.WITHDRAWING);
+    if (!this.claim) {
+      throw new ClaimNotSetError();
+    }
+    if ((await this.checkTransactionStatus(this.transactions.withdrawChallengeDepositTxn, ContractType.OUTBOX)) > 0) {
```

Comment:
(ignoreable) Consider extracting the await statement to a named variable for readability